src/store/apps.js

```diff
@@ -295,10 +295,10 @@ const actions = {
 		}).catch((error) => context.commit('API_FAILURE', { appId, error }))
 	},
 
-	uninstallApp(context, { appId }) {
+	uninstallApp(context, { appId, removeData }) {
 		return api.requireAdmin().then((response) => {
 			context.commit('startLoading', appId)
-			return api.get(generateUrl(`/apps/app_api/apps/uninstall/${appId}`))
+			return api.get(generateUrl(`/apps/app_api/apps/uninstall/${appId}/${removeData}`))
```

Comment:
if there is no strong reason to make it a path parameter, please, change it to query parameter

Reply:
As in the URL string should be `/apps/app_api/apps/uninstall/${appId}?removeData=${removeData}` instead?
That's the only other way I can think of; I don't quite understand how the `generateUrl` function is able to add query parameters otherwise (I thought it was the [`params`](https://nextcloud-libraries.github.io/nextcloud-router/functions/generateUrl.html) argument at first, but that turned out to be wrong).